dev_docs/tutorials/saved_objects.mdx

```diff
@@ -161,134 +162,29 @@ identify this reference. This guarantees that the id the reference points to alw
 
 ## Migrations
 
-Saved Objects support schema changes between Kibana versions, which we call migrations. Migrations are
- applied when a Kibana installation is upgraded from one version to a newer version, when exports are imported via
+Saved Objects support schema changes between Kibana versions, which we call migrations, implemented with model versions.
+Model version transitions are applied when a Kibana installation is upgraded from one version to a newer version, when exports are imported via
   the Saved Objects Management UI, or when a new object is created via the HTTP API.
 
-### Writing migrations
+### Defining model versions
 
-Each Saved Object type may define migrations for its schema. Migrations are specified by the Kibana version number, receive an input document,
- and must return the fully migrated document to be persisted to Elasticsearch.
+As for old migrations, model versions are bound to a given [savedObject type](https://github.com/elastic/kibana/blob/9b330e493216e8dde3166451e4714966f63f5ab7/packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts#L22-L27)
 
-Let’s say we want to define two migrations: - In version 1.1.0, we want to drop the subtitle field and append it to the title - In version
- 1.4.0, we want to add a new id field to every panel with a newly generated UUID.
+When registering a SO type, a new [modelVersions](https://github.com/elastic/kibana/blob/9a6a2ccdff619f827b31c40dd9ed30cb27203da7/packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts#L138-L177)
+property is available. This attribute is a map of [SavedObjectsModelVersion](https://github.com/elastic/kibana/blob/9a6a2ccdff619f827b31c40dd9ed30cb27203da7/packages/core/saved-objects/core-saved-objects-server/src/model_version/model_version.ts#L12-L20)
+which is the top-level type/container to define model versions.
 
-First, the current mappings should always reflect the latest or "target" schema. Next, we should define a migration function for each step in the schema evolution:
+This map follows a similar `{ [version number] => version definition }` format as the old migration map, however 
```

Comment:
Are we going to keep a reference around to the old migration map in the docs? If not, I'd consider not referencing it (just thinking of new developers reading this).

Maybe something like:

"The modelVersion map is of the form `{ [version number] => version definition }`, using single integer to identify a version definition"